ext/crypto/00_crypto.js

```diff
@@ -3226,6 +3230,103 @@
     }
   }
 
+  function exportKeyEC(format, key, innerKey) {
+    switch (format) {
+      case "pkcs8": {
+        // 1.
+        if (key[_type] !== "private") {
+          throw new DOMException(
+            "Key is not a private key",
+            "InvalidAccessError",
+          );
+        }
+
+        // 2.
+        const data = core.opSync("op_crypto_export_key", {
+          algorithm: key[_algorithm].name,
+          namedCurve: key[_algorithm].namedCurve,
+          format: "pkcs8",
+        }, innerKey);
+
+        // 3.
+        return data.buffer;
+      }
+      case "spki": {
+        // 1.
+        if (key[_type] !== "public") {
+          throw new DOMException(
+            "Key is not a public key",
+            "InvalidAccessError",
+          );
+        }
+
+        // 2.
+        const data = core.opSync("op_crypto_export_key", {
+          algorithm: key[_algorithm].name,
+          namedCurve: key[_algorithm].namedCurve,
+          format: "spki",
+        }, innerKey);
+
+        // 3.
+        return data.buffer;
+      }
+      case "jwk": {
+        // 1-2.
+        const jwk = {
+          kty: "EC",
+        };
+
+        // 3.
+        if (key[_algorithm].name == "ECDSA") {
```

Comment:
Based on the spec, a similar `switch` for ECDH should be there. It is possible to manipulate `key[_algorithm].namedCurve` by the user which in the current case will produce incorrect `jwk.crv`.

Reply:
For ECDH, jwk.alg is a constant ECDH so no `switch` needed I think. Will fix missing constant in next commit.

As for manipulating `key[_algorithm]`, manipulating this is possible all over the current implementation, for example,  `key[_algorithm].namedCurve` is used during `deriveBits` and `sign( ECDSA )`, or `key[_algorithm].hash.name`.

We could fix this in the general case by adding curve and hash info to Rust KeyData structures, but I think this is more of a major refactor. PLMK what you think.

Reply:
Alrighty It's fine then :+1: Separating ECDH and ECDSA impl (below comment) should fix the confusion